### PR TITLE
bpo-26131: deprecate the use of load_module()

### PR DESCRIPTION
**Pull request by brettcannon**
Deprecating the use of `load_module()` requires two parts:

1. The actual `load_module()` implementations are deprecated.
2. Use of `load_module()` as a fallback for `exec_module()` trigger an `ImportWarning`.

This makes sure that either direct use of the method or indirect use by the import system raises _some_ warning.

The method itself has been documented as deprecated since Python 3.6 and has been superceded by exec_module().

https://bugs.python.org/issue26131

### CODE REVIEW

File: Lib/importlib/_bootstrap.py [hunk 1]
```diff
@@ -620,9 +623,8 @@ def _exec(spec, module):
 
 
 def _load_backward_compatible(spec):
-    # (issue19713) Once BuiltinImporter and ExtensionFileLoader
-    # have exec_module() implemented, we can add a deprecation
-    # warning here.
+    # It is assumed that all callers have warned about using load_module()
```
Comment (warsaw) [inline Lib/importlib/_bootstrap.py]: "...have been..."?

**Review by warsaw — CHANGES_REQUESTED** [review]
One suggestion.  There are also test failures.

**Comment by brettcannon** [comment]
I have made the requested changes; please review again

**Comment by brettcannon** [comment]
This is going to require implementing `exec_module()` for `zipimport`.

**Pull request closed without merging**